com/amazonaws/athena/connectors/udfs/SkyflowUDFHandler.java

```diff
@@ -0,0 +1,129 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class SkyflowUDFHandler
+        extends UserDefinedFunctionHandler
+{
+    private static final String SOURCE_TYPE = "athena_common_udfs";
+    private final CachableSecretsManager cachableSecretsManager;
+    Boolean logLevel = Boolean.valueOf(System.getenv("LOG_LEVEL"));
+
+    public SkyflowUDFHandler()
+    {
+        this(new CachableSecretsManager(AWSSecretsManagerClient.builder().build()));
+    }
+    
+    SkyflowUDFHandler(CachableSecretsManager cachableSecretsManager)
+    {
+        super(SOURCE_TYPE);
+        this.cachableSecretsManager = cachableSecretsManager;
+    }
+   public String detokenize(String input, String role, String redactionParam)
+    {
+        String response = null;
+        try {
+            String vaultID = System.getenv("VAULT_ID");
+            String vaultURL = System.getenv("VAULT_URL");
+            String secretName = System.getenv("SECRET");
+            String secretString = cachableSecretsManager.getSecret(secretName);
+            String redaction = "DEFAULT";
+            if (redactionParam.length() > 0) {
+                redaction = redactionParam;
+            }
+            String endPointURL = vaultURL + "/v1/vaults/" + vaultID + "/detokenize";
+            JSONParser parser = new JSONParser();
+            JSONObject secrets = (JSONObject) parser.parse(secretString);
+            String apiKey = (String) secrets.get(role);
+            JSONObject bodyJson = new JSONObject();
+            JSONArray detokenizationParameters = new JSONArray();
+            JSONObject parameterObject = new JSONObject();
+            parameterObject.put("token", input);
+            parameterObject.put("redaction", redaction);
+            detokenizationParameters.add(parameterObject);
+            bodyJson.put("detokenizationParameters", detokenizationParameters);
+            Map<String, String> headers = new HashMap<>();
+            headers.put("Authorization", "Bearer " + apiKey);
+            long startTimeD = System.currentTimeMillis();
+            String apiResponse = HttpUtility.sendRequest("POST", new URL(endPointURL), bodyJson, headers);
+            long endTimeD = System.currentTimeMillis();
+            long latency = endTimeD - startTimeD;
+            if (logLevel) {
+                System.out.println("Skyflow Detokenize Latency " + latency + " ms");
+            }
+            JSONObject responseJson = (JSONObject) parser.parse(apiResponse);
+            JSONArray responseRecords = (JSONArray) responseJson.get("records");
+            JSONObject responseObject = (JSONObject) responseRecords.get(0);
+            response = (String) responseObject.get("value");
+        } 
+        catch (Exception e) {
+            System.out.println("Error" + e);
+            response = "error";
+        }
+        return response;
+    }
+
+    public String getrecord(String skyflowId, String tableName, String role, String redactionParam)
```

Comment:
Only a small style note, not a blocker: try to use camelCase for customer-facing function names. e.g. `getRecord`? Just for consistency across UDFs.

Reply:
Unfortunately, camelcase is not permitted in Athena UDFs. @jstjoe